js/background.js

```diff
@@ -429,27 +430,38 @@ function banImage(message) {
             bannedUrls[url] = { 'location' : message.location };
             update = true;
         }
-        if (update) localStorage.setItem('HoverZoomBannedUrls', JSON.stringify(bannedUrls));
+        if (update) {
+            localStorage.setItem('HoverZoomBannedUrls', JSON.stringify(bannedUrls));
+            // send updated list to tabs
+            sendBannedImages();
+        }
     } catch {}
 }
 
 // clear list of banned image, video or audio track urls
-function resetBannedImages() {   
-    localStorage.removeItem('HoverZoomBannedUrls');
+function resetBannedImages() {
+    localStorage.setItem('HoverZoomBannedUrls', '{}');
+    // send updated list to tabs
+    sendBannedImages();
 }
 
-// check if url of image, video or audio track belongs to ban list
-function isImageBanned(message) {
-    const url = message.url;
-    let bannedUrls = localStorage.getItem('HoverZoomBannedUrls') || '{}';
+// send list of banned image, video or audio track urls to all tabs
+function sendBannedImages() {
+    let list = localStorage.getItem('HoverZoomBannedUrls') || '{}';
     try {
-        bannedUrls = JSON.parse(bannedUrls);
-    } catch { return false; }
-    if (!bannedUrls[url]) {
-        return false;
-    } else {
-        return true;
-    }
+        list = JSON.parse(list);
+    } catch { return; }
+
+    var request = {action:'bannedImagesListChanged', 'list':list};
+    chrome.windows.getAll(null, function (windows) {
+        for (var i = 0; i < windows.length; i++) {
+            chrome.tabs.query({windowId: windows[i].id}, function (tabs) {
+                for (var j = 0; j < tabs.length; j++) {
+                    chrome.tabs.sendMessage(tabs[j].id, request);
+                }
+            });
+        }
+    });
```

Comment:
This block is duplicated in `sendOptions()` function. It would be best to rename that one to something like `broadcastMessage` instead of duplicating the code.